### ISSUE TITLE
SSL verification properties are not taken into account when fetching files from OpenCTI

### ISSUE THREAD

**Issue by romain-filigran** [OpenCTI-Platform/client-python#682]
## Description

SSL verification properties are not handle in "fetch_opencti_file" method.
https://github.com/OpenCTI-Platform/client-python/blob/master/pycti/api/opencti_api_client.py#L380
This prevent "import-document" and "import-file-stix" connectors to work with self-signed certificates or invalid certificates.

## Environment

1. pycti version: 6.1.11

## Additional information

see: https://github.com/OpenCTI-Platform/opencti/issues/7349

**Comment by helene-nguyen** [comment]
I was able to reproduce the issue with self-signed certificate:

```sh
{"timestamp": "2024-06-18T12:51:54.632685Z", "level": "ERROR", "name": "ImportFileStix", "message": "Error in message processing, reporting error to API", "exc_info": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py\", line 466, in _make_request\n    self._validate_conn(conn)\n  
File \"/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py\", line 1095, in _validate_conn\n    conn.connect()\n  
File \"/usr/local/lib/python3.11/site-packages/urllib3/connection.py\", line 652, in connect\n    sock_and_verified = _ssl_wrap_socket_and_match_hostname(\n                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n 
File \"/usr/local/lib/python3.11/site-packages/urllib3/connection.py\", line 805, in _ssl_wrap_socket_and_match_hostname\n    ssl_sock = ssl_wrap_socket(\n               ^^^^^^^^^^^^^^^^\n 
File \"/usr/local/lib/python3.11/site-packages/urllib3/util/ssl_.py\", line 465, in ssl_wrap_socket\n    ssl_sock = _ssl_wrap_socket_impl(sock, context, tls_in_tls, server_hostname)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  
File \"/usr/local/lib/python3.11/site-packages/urllib3/util/ssl_.py\", line 509, in _ssl_wrap_socket_impl\n    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  
File \"/usr/local/lib/python3.11/ssl.py\", line 517, in wrap_socket\n    return self.sslsocket_class._create(\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  
File \"/usr/local/lib/python3.11/ssl.py\", line 1104, in _create\n    self.do_handshake()\n  
File \"/usr/local/lib/python3.11/ssl.py\", line 1382, in do_handshake\n    self._sslobj.do_handshake()\nssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:1006)\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  
File \"/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py\", line 789, in urlopen\n    response = self._make_request(\n               ^^^^^^^^^^^^^^^^^^^\n  
File \"/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py\", line 490, in _make_request\n    raise new_e\nurllib3.exceptions.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:1006)\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  
File \"/usr/local/lib/python3.11/site-packages/requests/adapters.py\", line 667, in send\n    resp = conn.urlopen(\n           ^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py\", line 843, in urlopen\n    retries = retries.increment(\n              ^^^^^^^^^^^^^^^^^^\n  
File \"/usr/local/lib/python3.11/site-packages/urllib3/util/retry.py\", line 519, in increment\n    raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]\n
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nurllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='opencti', port=8080): Max retries exceeded with url: /storage/get/import/Artifact/763a3bc7-1fec-4d7e-a938-aafa5115de03/2024-06-17T09_54_52.971Z_TLP_ALL_(ExportFileStix2)_Report-Ransomware%20Attackers%20May%20Have%20Used%20Privilege%20Escalation%20Vulnerability%20as%20Zero-day_full.json 
(Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:1006)')))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/site-packages/pycti/connector/opencti_connector_helper.py\", line 349, in _data_handler\n    message = self.callback(event_data)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^\n  
File \"/opt/opencti-connector-import-file-stix/import-file-stix.py\", line 31, in _process_message\n    file_content = self.helper.api.fetch_opencti_file(file_uri)\n                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  
File \"/usr/local/lib/python3.11/site-packages/pycti/api/opencti_api_client.py\", line 380, in fetch_opencti_file\n    r = self.session.get(fetch_uri, headers=self.request_headers)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  
File \"/usr/local/lib/python3.11/site-packages/requests/sessions.py\", line 602, in get\n    return self.request(\"GET\", url, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  
File \"/usr/local/lib/python3.11/site-packages/requests/sessions.py\", line 589, in request\n    resp = self.send(prep, **send_kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  
File \"/usr/local/lib/python3.11/site-packages/requests/sessions.py\", line 703, in send\n    r = adapter.send(request, **kwargs)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/site-packages/requests/adapters.py\", line 698, in send\n    raise SSLError(e, request=request)\nrequests.exceptions.SSLError: HTTPSConnectionPool(host='opencti', port=8080): Max retries exceeded with url: /storage/get/import/Artifact/763a3bc7-1fec-4d7e-a938-aafa5115de03/2024-06-17T09_54_52.971Z_TLP_ALL_(ExportFileStix2)_Report-Ransomware%20Attackers%20May%20Have%20Used%20Privilege%20Escalation%20Vulnerability%20as%20Zero-day_full.json 
(Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:1006)')))"}

{"timestamp": "2024-06-18T12:51:54.633240Z", "level": "INFO", "name": "api", "message": "Reporting work update_processed", "attributes": {"work_id": "work_import-file-stix--test_2024-06-18T12:51:54.479Z"}}

/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py:1099: InsecureRequestWarning: Unverified HTTPS request is being made to host 'opencti'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  warnings.warn(
{"timestamp": "2024-06-18T12:51:55.616088Z", "level": "INFO", "name": "ImportFileStix", "message": "Message processed, thread terminated", "attributes": {"tag": 1}}
```

I'm on it :)

**Comment by helene-nguyen** [comment]
Will be fixed by #684
